src/main/scala/org/broadinstitute/dsde/workbench/sam/api/StandardUserInfoDirectives.scala

```diff
@@ -1,15 +1,38 @@
 package org.broadinstitute.dsde.workbench.sam.api
 
 import akka.http.scaladsl.server._
-import akka.http.scaladsl.server.Directives.headerValueByName
-import org.broadinstitute.dsde.workbench.model.{WorkbenchUserEmail, WorkbenchUserId}
+import akka.http.scaladsl.server.Directives.{headerValueByName, onSuccess}
+import org.broadinstitute.dsde.workbench.model._
 import org.broadinstitute.dsde.workbench.sam.model.UserInfo
 
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+
+
 trait StandardUserInfoDirectives extends UserInfoDirectives {
+  val petSAdomain = "\\S+@\\S+.iam.gserviceaccount.com"
```

Comment:
I think you want to escape the dots so they're treated as literal dots, not regex matchers. E.g., right now this matches the following which is not correct:
```
scala> "abc@fooXiamYgserviceaccountZcom".matches(petSAdomain)
res7: Boolean = true
```

This fixes it:
```
scala> val petSAdomain = "\\S+@\\S+\\.iam\\.gserviceaccount\\.com"
petSAdomain: String = \S+@\S+\.iam\.gserviceaccount\.com

scala> "abc@fooaiamagserviceaccountacom".matches(petSAdomain)
res8: Boolean = false

scala> "abc@foo.iam.gserviceaccount.com".matches(petSAdomain)
res9: Boolean = true
```